p2p/swarm.go

```diff
@@ -160,20 +218,19 @@ func (s *swarm) Start() error {
 		}()
 	}
 
-	if s.config.SwarmConfig.Gossip {
-		go func() {
-			if s.config.SwarmConfig.Bootstrap {
-				s.waitForBoot()
-			}
-			err := s.gossip.Start()
-			if err != nil {
-				s.gossipErr = err
-				s.Shutdown()
-			}
-			<-s.gossip.Initial()
-			close(s.gossipC)
-		}() // todo handle error async
-	}
+	// wait for neighborhood
+	go func() {
+		if s.config.SwarmConfig.Bootstrap {
+			s.waitForBoot()
+		}
+		err := s.StartNeighborhood()
```

Comment:
Before you start the gathering peers you need to start all the relevant protocol (gossip, sync) so that they will start collecting peers from the beginning